### DISCUSSION TITLE
Fastlane gym can export ".app" file ?

### DISCUSSION THREAD

**Discussion by uthens** [fastlane/fastlane#21907] [General]
I know fastlane gym can export to ".ipa" file, but on my team we want to used .app file for automated testing tool. 

How to use fastlane gym or other action on fastlane export to ".app" file ? 

Best,
UT

**Comment by uthens** [comment]
https://docs.fastlane.tools/actions/gym/
"gym builds and packages iOS apps for you. It takes care of all the heavy lifting and makes it super easy to generate a signed ipa or app file"

**Comment by Patricio879** [comment]
Yes, Fastlane's "gym" tool can export a ".app" file. When you use Fastlane's gym tool, it compiles your Xcode project and generates an .ipa file by default, which contains your app bundle along with other necessary files for deployment. However, within the .ipa file, the .app bundle is included.

If you specifically need the .app bundle without the additional packaging of the .ipa file, you can extract it from the .ipa file generated by gym. There are various tools available to extract the .app bundle from an .ipa file, such as unzip or third-party utilities.

So, while gym primarily generates .ipa files, the .app bundle, which is the core of the application, is included within that .ipa file and can be extracted if needed.
